### PR TITLE
#2644 Password protected message compliance

### LINKED ISSUE #2644: Password-protected messages compliance
To ensure password-protected messages comply with security rules, client apps must check the subjects of composed password-protected email messages for specified strings. If any configured terms are found, the application should display an error message indicating that the password encryption method is incompatible with the composed message.

Client configuration will have 2 new optional properties (subject check should be performed only if both properties are set):
- `disallow_password_messages_for_terms` (_string[]_) - specifies an array of strings to check against the subject of the composed password-protected message. If any string in this array is found in the subject, an error alert must be displayed.
- `disallow_password_messages_error_text` (_string_) - defines the text to be displayed in the error alert.

related https://github.com/FlowCrypt/flowcrypt-browser/issues/5878

### PR DESCRIPTION
**Pull request by ioanmo226**
This PR implemented password protected message compliance

close #2644// if this PR closes an issue

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCrypt/Models/Realm Models/ClientConfigurationRealmObject.swift [hunk 1]
```diff
@@ -19,6 +19,8 @@ final class ClientConfigurationRealmObject: Object {
     @Persisted var disallowAttesterSearchForDomains: Data?
     @Persisted var enforceKeygenAlgo: String?
     @Persisted var enforceKeygenExpireMonths: Int
+    @Persisted var disallowPasswordMessagesForTerms: Data?
+    @Persisted var disallowPasswordMessagesErrorText: String?
```
Comment (sosnovsky) [inline FlowCrypt/Models/Realm Models/ClientConfigurationRealmObject.swift:23]: looks like migration is missing, as I've got such error on app startup:

File: FlowCryptCommon/Extensions/StringExtensions.swift [hunk 2]
```diff
@@ -158,4 +158,30 @@ public extension String {
...
+            let lowerCaseTerm = term.lowercased()
+
+            // Check for exact matches (full-term match within the subject)
+            if lowerCaseSubject.contains(lowerCaseTerm),
+               lowerCaseSubject == lowerCaseTerm ||
+               lowerCaseSubject.hasPrefix(lowerCaseTerm + " ") ||
+               lowerCaseSubject.hasSuffix(" " + lowerCaseTerm) ||
+               lowerCaseSubject.contains(" " + lowerCaseTerm + " ") {
+                return false
+            }
```
Comment (sosnovsky) [inline FlowCryptCommon/Extensions/StringExtensions.swift:182]: after fixing it in the browser extension, let's update detection code here too

File: appium/tests/screenobjects/new-message.screen.ts [hunk 3]
```diff
@@ -203,6 +213,11 @@ class NewMessageScreen extends BaseScreen {
     await this.setSubject(subject);
   };
 
+  checkCustomAlertMessage = async (content: string) => {
+    console.log(content);
+    const customAlertMessageLabel = await this.customAlertMessageLabel;
+    console.log(await customAlertMessageLabel.getText());
+  };
```
[the remark this answers is not in the document]
Comment (ioanmo226) [reply appium/tests/screenobjects/new-message.screen.ts:220]: Oops! Missed it

File: FlowCrypt/Models/Realm Models/ClientConfigurationRealmObject.swift (hunk 1 above)
Comment (ioanmo226) [reply FlowCrypt/Models/Realm Models/ClientConfigurationRealmObject.swift:23]: Yep, Right

File: FlowCryptUI/Nodes/CustomAlertNode.swift [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        super.init()
+        setupNodes()
+    }
+
+    private func createNodes() {
+        overlayNode = createOverlayNode()
+        contentView = createContentView()
+        separatorNode = createSeparatorNode()
+        titleLabel = createTextNode(text: title, isBold: true, fontSize: 17)
+        messageLabel = createTextNode(text: message ?? "", isBold: false, fontSize: 13, identifier: "aid-custom-alert-message", detectLinks: true)
```
Comment (sosnovsky) [inline FlowCryptUI/Nodes/CustomAlertNode.swift:42]: let's center message text, to make it look like in default iOS alerts:

Screenshot 2024-12-11 at 12 47 15

Screenshot 2024-12-11 at 12 46 54

and is it possible to highlight links by different text color, not by underlining?
it seems to be more common on iOS

Comment (ioanmo226) [reply FlowCryptUI/Nodes/CustomAlertNode.swift:42]: Okay, got it

**Review by sosnovsky — APPROVED** [review]
All good now, thanks!

**Pull request merged**